docs-developer/custom-importer.md

```diff
@@ -0,0 +1,76 @@
+ * [existing importers](../src/profile-logic/import)
+
+## How to write a profile converter
+
+The rest of this guide will assume you are targeting the processed profile format. From a high level point of view, the profile is broken down into the basic profile, with meta information, and a list of threads. These threads are grouped in the UI by their PID. The UI also dynamically generates a few "tracks" from other data sources, like the memory track or screenshot track. This guide primarily documents working with threads.
+
+A good place to start would probably be to view some of the existing [blank profile generating functions](../src/profile-logic/data-structures.js).
+
+## Some concepts used in the data structures
+
+The profile format uses a lot of indexes into other data structures. For instances, this is how you would extract out a function name from a profile. (This should work from the web console when viewing a profile on profiler.firefox.com.)
+
+```js
+{
+  // Get the first sample out of the first thread:
+  const threadIndex = 0;
+  const sampleIndex = 0;
+
+  const thread = profile.threads[threadIndex];
+
+  // Look up the stack index from the samples table.
+  const stackIndex = thread.samples.stack[sampleIndex];
+  console.log({stackIndex})
+
+  // Look up the frame from the stack table.
+  const frameIndex = thread.stackTable.frame[stackIndex];
+  console.log({frameIndex})
+
+  // Look up the function from the frame table.
+  const funcIndex = thread.frameTable.func[frameIndex];
+  console.log({funcIndex})
+
+  // Look up the string index from the func table.
+  const stringIndex = thread.funcTable.name[funcIndex];
+  console.log({stringIndex})
+
+  console.log(
+    'Function name of the first sample:',
+    thread.stringTable.getString(stringIndex)
+  );
+}
+```
+
+It is probably to read up some on profile format docs for more information on how this structure is used.
+
+## Tips
+
+ * Use `{ name: 'GeckoMain' }` for the main thread.
+ * Ensure that the `pid` value is numeric, and points to the proper threads to nest threads in the timeline.
+ * Processed profiles have their timestamps adjusted so that all processes use the same timeline.
+ * Gecko profiles adjust the marker and sample start times based on when the process was created (based off of `geckoProfile.meta.startTime`).
+
+## Samples
+
+The call tree, flame graph, and stack chart all use the samples data to generate the reports. If your data is shaped like sample data, it can be easy to shove the information in there. At the time of this writing we don't support variable durations. It's assumed 1 samples is equal in length to the sampling interval. This is probably going to change soon. Most likely the profile formats will assume that there is at least one sample in a thread. It might be necessary to fill out the samples with dummy time data if your format isn't needing samples. The sample data should match the time range as the marker data.
```

Comment:
Maybe insert some new lines in this big paragraph.

For the last sentence you can say it's good enough to push 1 sample for the start and 1 sample for the end.